napari/_vispy/visuals/volume.py

```diff
@@ -35,7 +35,7 @@
 vec4 calculateCategoricalColor(vec4 betterColor, vec3 loc, vec3 step)
 {
     // Calculate color by incorporating ambient and diffuse lighting
-    vec4 color0 = $sample(u_volumetex, loc);
+    vec4 color0 = $get_data(loc);
```

Comment:
I don't fully understand these changes, but I want to learn! I understand that these changes make our custom shader consistent with the changes in https://github.com/vispy/vispy/pull/2322

What was `sample` doing previously? Looks like it was an alias to the texture's `glsl_sample` method. I'm guessing that was handling 3d interp previously? And now we're just getting the color data and it's being handled elsewhere in the rendering pipeline (e.g. another shader)?

Reply:
`$sample` was basically calling the hardware texture lookup. The new `$get_data` is named to reflect the equivalent in `ImageVisual`, and effectively calls the shader code inside `_INTERPOLATION_TEMPLATE` with whatever interpolation function is chosen (the actual shader for the intepolation lives in https://github.com/vispy/vispy/blob/main/vispy/glsl/misc/spatial-filters.frag).

Reply:
Thanks for the explanation!